diff -urN mod_caml-1.3.4-orig/Makefile mod_caml-1.3.4-corrected/Makefile
--- mod_caml-1.3.4-orig/Makefile	2004-11-20 00:12:14.000000000 +0100
+++ mod_caml-1.3.4-corrected/Makefile	2004-12-05 21:23:54.445456776 +0100
@@ -25,16 +25,16 @@
 SED := sed
 
 CC	:= $(CC)
-CFLAGS	:= -I $(APACHEINCDIR) -DEAPI
+CFLAGS	:= -I $(APACHEINCDIR) -I $(APRINCDIR) -I $(APRUTILINCDIR) -DEAPI
 
-OCAMLCINCS := -I $(PCRELIBDIR) -I $(APACHEINCDIR)
+OCAMLCINCS := -I $(PCRELIBDIR) -I $(APACHEINCDIR) 
 ALL_CMOS := apache.cmo mod_caml_config.cmo mod_caml.cmo cgi_escape.cmo \
 	template.cmo cgi.cmo
 ALL_CMXS := cgi_escape.cmx template.cmx
 
 OCAMLCFLAGS := -g -w s $(OCAMLCINCS)
 OCAMLOPTFLAGS := -w s $(OCAMLCINCS)
-OCAMLLIBS += -lcamlrun -ltermcap -lunix -lstr 
+OCAMLLIBS += -lcamlrun -lunix -lstr 
 ALL_CMAS += dynlink.cma str.cma pcre.cma unix.cma
 
 OCAMLDOCFLAGS := -html -stars -sort $(OCAMLCINCS)
@@ -176,8 +176,9 @@
 	$(INSTALL)    -m 0755 -d $(DESTDIR)$(APACHELIBDIR)
 	$(INSTALL)    -m 0755 -d $(DESTDIR)$(OCAMLAPACHEDIR)
 	$(INSTALL)    -m 0755 -d $(DESTDIR)$(MODCAMLLIBDIR)
+	$(INSTALL)    -m 0755 -d $(DESTDIR)$(APACHECONFDIR)
 	$(INSTALL) -c -m 0755 mod_caml.so $(DESTDIR)$(APACHELIBDIR)
-	$(INSTALL) -c -m 0644 500mod_caml.info $(DESTDIR)$(APACHELIBDIR)
+	$(INSTALL) -c -m 0644 500mod_caml.info $(DESTDIR)$(APACHECONFDIR)/500mod_caml.conf
 	$(INSTALL) -c -m 0644 mod_caml_config.cmi $(DESTDIR)$(OCAMLAPACHEDIR)
 	$(INSTALL) -c -m 0644 apache.mli apache.cmi $(DESTDIR)$(OCAMLAPACHEDIR)
 	$(INSTALL) -c -m 0644 mod_caml.mli mod_caml.cmi \
@@ -232,9 +233,6 @@
 install-doc:
 	$(INSTALL)    -m 0755 -d $(DESTDIR)$(DOCDIR)
 	$(INSTALL) -c -m 0644 AUTHORS FAQ README TODO $(DESTDIR)$(DOCDIR)
-	$(INSTALL) -c -m 0644 \
-	  examples/simple-scripts/*.ml \
-	  $(DESTDIR)$(DOCDIR)
 
 install-html:
 	$(INSTALL)    -m 0755 -d $(DESTDIR)$(DOCDIR)/html
@@ -242,7 +240,7 @@
 
 install-conf:
 	$(INSTALL)    -m 0755 -d $(DESTDIR)$(APACHECONFDIR)
-	$(INSTALL) -c -m 0644 modcaml-example.conf $(DESTDIR)$(APACHECONFDIR)
+	$(INSTALL) -c -m 0644 modcaml-example.conf $(DESTDIR)$(DOCDIR)/
 
 dist:
 	$(MAKE) check-manifest
diff -urN mod_caml-1.3.4-orig/Makefile.config mod_caml-1.3.4-corrected/Makefile.config
--- mod_caml-1.3.4-orig/Makefile.config	2004-11-20 00:12:14.000000000 +0100
+++ mod_caml-1.3.4-corrected/Makefile.config	2004-12-05 21:26:59.182372496 +0100
@@ -25,8 +25,8 @@
 # For Apache 1.3, set this to 1
 # For Apache 2.x, set this to 2
 
-APACHE_MAJOR := 1
-#APACHE_MAJOR := 2
+#APACHE_MAJOR := 1
+APACHE_MAJOR := 2
 
 # APXS
 # For Apache 1.3, you need to use apxs or /full/path/to/apxs if it's not on
@@ -42,8 +42,8 @@
 # APACHELIBDIR
 # This is where we're going to install mod_caml.so.
 
-APACHELIBDIR := /usr/lib/apache/1.3
-#APACHELIBDIR := /usr/lib/apache2/modules
+#APACHELIBDIR := /usr/lib/apache/1.3
+APACHELIBDIR := /usr/lib/apache
 #APACHELIBDIR := $(shell pwd)
 #APACHELIBDIR := /usr/local/apache/libexec
 #APACHELIBDIR := /usr/lib/mod_caml
@@ -52,7 +52,9 @@
 # APACHEINCDIR
 # Location of the Apache includes.
 
-APACHEINCDIR := /usr/include/apache-1.3
+APACHEINCDIR := /usr/include/apache
+APRINCDIR := /usr/include/apr
+APRUTILINCDIR := /usr/include/apr-util
 
 # MODCAMLLIBDIR
 # This is where we're going to install the default mod_caml handlers
@@ -64,7 +66,7 @@
 # APACHECONFDIR
 # Apache configuration directory.
 
-APACHECONFDIR := /etc/apache
+APACHECONFDIR := /etc/httpd/httpd.conf/
 
 # OCAMLC, OCAMLOPT, OCAMLDOC
 # The location of the OCaml compiler and tools. The defaults should be OK.
@@ -106,7 +108,7 @@
 # DOCDIR
 # This is where we will install the documentation.
 
-DOCDIR = /usr/share/doc/$(PACKAGE)-$(VERSION)
+DOCDIR = /usr/share/doc/apache-$(PACKAGE)-$(VERSION)
 
 # HAVE_POSTGRES
 # Tests whether you have (the OCaml module) Postgres installed.  If so, then
